### PR TITLE
Merge release/v6.1.0 into master

### PR DESCRIPTION
**Pull request by lucianomartin**

### CODE REVIEW

File: AN00209_xCORE-200_DSP_Library/app_adaptive/Makefile [hunk 1]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_adaptive/Makefile:4]: I thought that we agreed to using `ifeq ($(XCOREAI),1)` with the corresponding clause building for the XCORE-AI Explorer and the else clause building for XCORE-200 Explorer.

File: AN00209_xCORE-200_DSP_Library/app_atan2_hypot/Makefile [hunk 2]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_atan2_hypot/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_design/Makefile [hunk 3]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_design/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_fft_dif/Makefile [hunk 4]
```diff
@@ -1,13 +1,24 @@
+BUILD_FLAGS = -O2 -g -report
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_fft_dif/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_fft_double_buf/Makefile [hunk 5]
```diff
@@ -1,18 +1,19 @@
+COMMON_FLAGS = -O2 -g -report
+
+BUILD_FLAGS_complex = $(COMMON_FLAGS) -DNUM_CHANS=3
+BUILD_FLAGS_tworeals = $(COMMON_FLAGS) -DTWOREALS=1 -DNUM_CHANS=4
+
+BUILD_FLAGS_complex_int16_buf = $(COMMON_FLAGS) -DINT16_BUFFERS=1 -DNUM_CHANS=6
+BUILD_FLAGS_tworeals_int16_buf = $(COMMON_FLAGS) -DTWOREALS=1 -DINT16_BUFFERS=1 -DNUM_CHANS=8
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_fft_double_buf/Makefile:10]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_fft_real_single/Makefile [hunk 6]
```diff
@@ -1,7 +1,18 @@
+BUILD_FLAGS = -O2 -g -report
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_fft_real_single/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_fft_timing/Makefile [hunk 7]
```diff
@@ -1,7 +1,18 @@
+BUILD_FLAGS = -O2 -g -report
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_fft_timing/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_filters/Makefile [hunk 8]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_filters/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_math/Makefile [hunk 9]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_math/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_matrix/Makefile [hunk 10]
```diff
@@ -1,8 +1,25 @@
+BUILD_FLAGS = -O3 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_matrix/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_statistics/Makefile [hunk 11]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_statistics/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_vector/Makefile [hunk 12]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_vector/Makefile:4]: Same comment as app_adaptive Makefile.

File: AN00209_xCORE-200_DSP_Library/app_window_post_fft/Makefile [hunk 13]
```diff
@@ -1,7 +1,18 @@
+BUILD_FLAGS = -O2 -g -report
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline AN00209_xCORE-200_DSP_Library/app_window_post_fft/Makefile:4]: Same comment as app_adaptive Makefile.

File: tests/test_fft_forward_real/Makefile [hunk 14]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g -report -DDEBUG_PRINT_ENABLE=1
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline tests/test_fft_forward_real/Makefile:4]: Same comment as app_adaptive Makefile.

File: tests/test_fft_inverse_blank_forward/Makefile [hunk 15]
```diff
@@ -1,22 +1,32 @@
+BUILD_FLAGS = -O2 -g -report -DDEBUG_PRINT_ENABLE=1
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
Comment (mbanth) [inline tests/test_fft_inverse_blank_forward/Makefile:4]: Same comment as app_adaptive Makefile.

**Review by mbanth — COMMENTED** [review]
I noted a pendantic issue reguarding the use of conditionals in make files.  I have an underlying concern about how make files set the value of target-specific symbols once we have more than two target platforms.  Changing the make files as I have suggested is a step toward a more comprehensive solution, but it's not complete in itself.

File: AN00209_xCORE-200_DSP_Library/app_adaptive/Makefile (hunk 1 above)
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_adaptive/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_bfp/Makefile [hunk 16]
```diff
@@ -1,7 +1,18 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
[the remark this answers is not in the document]
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_bfp/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_complex/Makefile [hunk 17]
```diff
@@ -1,7 +1,18 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
[the remark this answers is not in the document]
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_complex/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_statistics/Makefile (hunk 11 above)
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_statistics/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_window_post_fft/Makefile (hunk 13 above)
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_window_post_fft/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: tests/test_fft_forward_real/Makefile (hunk 14 above)
Comment (lucianomartin) [reply tests/test_fft_forward_real/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: tests/test_fft_inverse_blank_forward/Makefile (hunk 15 above)
Comment (lucianomartin) [reply tests/test_fft_inverse_blank_forward/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_vector/Makefile (hunk 12 above)
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_vector/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_complex_fir/Makefile [hunk 18]
```diff
@@ -1,7 +1,18 @@
+BUILD_FLAGS = -O2 -g
+
+XCOREAI ?= 0
+ifeq ($(XCOREAI),0)
```
[the remark this answers is not in the document]
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_complex_fir/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

File: AN00209_xCORE-200_DSP_Library/app_atan2_hypot/Makefile (hunk 2 above)
Comment (lucianomartin) [reply AN00209_xCORE-200_DSP_Library/app_atan2_hypot/Makefile]: This has been fixed by https://github.com/xmos/lib_dsp/pull/193

**Review by mbanth — APPROVED** [review]
Looks good.

**Pull request merged**